energy-integration/farm-boosted-yields/src/lib.rs

```diff
@@ -98,4 +85,95 @@ pub trait FarmBoostedYieldsModule:
 
     #[storage_mapper("accumulatedRewardsForWeek")]
     fn accumulated_rewards_for_week(&self, week: Week) -> SingleValueMapper<BigUint>;
+
+    #[storage_mapper("farmClaimProgress")]
+    fn farm_claim_progress(
+        &self,
+        user: &ManagedAddress,
+        token_nonce: Nonce,
+    ) -> SingleValueMapper<ClaimProgress<Self::Api>>;
+}
+
+pub struct FarmBoostedYieldsWrapper<T: FarmBoostedYieldsModule> {
+    pub current_farm_token_nonce: Nonce,
+    pub phantom: PhantomData<T>,
+}
+
+impl<T> WeeklyRewardsSplittingTraitsModule for FarmBoostedYieldsWrapper<T>
+where
+    T: FarmBoostedYieldsModule,
+{
+    type WeeklyRewardsSplittingMod = T;
+
+    fn get_current_farm_token_nonce(&self) -> u64 {
+        self.current_farm_token_nonce
+    }
+
+    // TODO - token identifier?
```

Comment:
Can be gotten via the reward_token_id storage_mapper, as you have access to it here.

Reply:
Done.